packages/langium/src/syntax-tree.ts

```diff
@@ -95,10 +95,21 @@ export interface LinkingError extends ReferenceInfo {
 export interface AstReflection {
     getAllTypes(): string[]
     getReferenceType(referenceId: string): string
-    isInstance(node: AstNode, type: string): boolean
+    getTypeMetaData(type: string): TypeMetaData
+    isInstance(node: unknown, type: string): boolean
     isSubtype(subtype: string, supertype: string): boolean
 }
 
+export interface TypeMetaData {
+    name: string
+    mandatory: TypeMandatoryField[]
+}
+
+export interface TypeMandatoryField {
```

Comment:
Please add documentation to the new types, especially what 'mandatory' means here.